sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -541,15 +541,17 @@ object IntervalUtils {
   }
 
   private val signRe = "(?<sign>[+|-])"
-  private val dayRe = "(?<day>\\d+)"
+  private val dayRe = "(?<day>\\d{1,9})"
   private val hourRe = "(?<hour>\\d{1,2})"
   private val minuteRe = "(?<minute>\\d{1,2})"
   private val secondRe = "(?<second>(\\d{1,2})(\\.(\\d{1,9}))?)"
+  private val hourBoundRe = "(?<hour>\\d{1,10})"
+  private val minuteBoundRe = "(?<minute>\\d{1,12})"
 
   private val dayTimePattern = Map(
-    (MINUTE, SECOND) -> s"^$signRe?$minuteRe:$secondRe$$".r,
-    (HOUR, MINUTE) -> s"^$signRe?$hourRe:$minuteRe$$".r,
-    (HOUR, SECOND) -> s"^$signRe?$hourRe:$minuteRe:$secondRe$$".r,
+    (MINUTE, SECOND) -> s"^$signRe?$minuteBoundRe:$secondRe$$".r,
+    (HOUR, MINUTE) -> s"^$signRe?$hourBoundRe:$minuteRe$$".r,
+    (HOUR, SECOND) -> s"^$signRe?$hourBoundRe:$minuteRe:$secondRe$$".r,
     (DAY, HOUR) -> s"^$signRe?$dayRe $hourRe$$".r,
     (DAY, MINUTE) -> s"^$signRe?$dayRe $hourRe:$minuteRe$$".r,
     (DAY, SECOND) -> s"^$signRe?$dayRe $hourRe:$minuteRe:$secondRe$$".r
```

Comment:
All these duplicate https://github.com/apache/spark/blob/26d1bb16bc565dbcb1a3f536dc78cd87be6c2468/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala#L248-L291
We need to duplicate the code.